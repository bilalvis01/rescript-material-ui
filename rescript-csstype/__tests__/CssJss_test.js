// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Jss = require("jss").default;
var Curry = require("rescript/lib/js/curry.js");
var Jss$Ress = require("../bindings/Jss.js");
var Belt_Option = require("rescript/lib/js/belt_Option.js");
var CssColor$Ress = require("../src/property/CssColor.js");
var JssPresetDefault = require("jss-preset-default").default;

Jss.setup(JssPresetDefault());

var data = {
  space: 24,
  color: /* Blue */0
};

test("jss", (function () {
        var s = Curry._1(Jss$Ress.Css.styles, [
              Curry._2(Jss$Ress.Css.rule, "app", [
                    Curry._2(Jss$Ress.Css.rule, "& .wrapper", [
                          Curry._8(Jss$Ress.Css.background, undefined, undefined, undefined, undefined, undefined, undefined, undefined, Curry._1(Jss$Ress.Css.url, "image.png")),
                          Curry._1(Jss$Ress.Css.color, Curry._3(Jss$Ress.Css.rgb, 255, 255, 255)),
                          Curry._1(Jss$Ress.Css.paddingTop, Curry._1(Jss$Ress.Css.px, 24)),
                          Curry._1(Jss$Ress.Css.paddingBottom, Curry._1(Jss$Ress.Css.px, 24)),
                          Curry._1(Jss$Ress.Css.paddingLeft, Curry._1(Jss$Ress.Css.px, 40)),
                          Curry._1(Jss$Ress.Css.paddingRight, Curry._1(Jss$Ress.Css.px, 40)),
                          Curry._1(Jss$Ress.Css.marginNumber, 24)
                        ]),
                    Curry._2(Jss$Ress.Css.rule, "& .button", [
                          Curry._1(Jss$Ress.Css.color, Curry._4(Jss$Ress.Css.hsla, 360, Curry._1(Jss$Ress.Css.pct, 100), Curry._1(Jss$Ress.Css.pct, 50), 0.5)),
                          Curry._1(Jss$Ress.Css.hover, [Curry._8(Jss$Ress.Css.background, undefined, undefined, undefined, undefined, undefined, undefined, undefined, "blue")])
                        ]),
                    Curry._1(Jss$Ress.Css.hover, [Curry._1(Jss$Ress.Css.color, "blue")])
                  ]),
              Curry._2(Jss$Ress.Css.rule, "header", [
                    Curry._1(Jss$Ress.Css.important, Curry._1(Jss$Ress.Css.colorFn, (function (data) {
                                return Belt_Option.map(data.color, (function (color) {
                                              if (color) {
                                                return CssColor$Ress.value("red");
                                              } else {
                                                return CssColor$Ress.value("blue");
                                              }
                                            }));
                              }))),
                    Curry._1(Jss$Ress.Css.marginFn, (function (data) {
                            return Belt_Option.map(data.space, (function (space) {
                                          return space;
                                        }));
                          }))
                  ])
            ]);
        expect(Jss.createStyleSheet(s).update(data).toString()).toMatchSnapshot();
        
      }));

exports.data = data;
/*  Not a pure module */
