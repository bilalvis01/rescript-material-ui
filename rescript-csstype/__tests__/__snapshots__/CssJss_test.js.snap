// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jss 1`] = `
".app-0-0-1 .wrapper {
  color: rgb(255, 255, 255);
  margin: 24px;
  background: url(\\"image.png\\");
  padding-top: 24px;
  padding-left: 40px;
  padding-right: 40px;
  padding-bottom: 24px;
}
.app-0-0-1 .button {
  color: hsla(360, 100%, 50%, 0.5);
}
.app-0-0-1:hover {
  color: blue;
}
.app-0-0-1 .button:hover {
  background: blue;
}
.header-0-0-2 {
  color: blue !important;
  margin: 24px;
}"
`;
